src/containers/Stories/components/TagsFilter.js

```diff
@@ -1,48 +1,59 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useMemo } from 'react';
 
-import { useAllTags, useDisplayedTags } from 'containers/Stories/storiesHooks';
+import { useTags } from 'containers/Stories/storiesHooks';
 import TagFilter from 'components/TagFilter';
 import { StoriesList } from 'containers/Stories/components/StoriesList';
 import { useTranslation } from 'react-i18next';
 
-export const TagsFilter = ({ changeLocationByPath }) => {
+export const TagsFilter = ({ changeLocationByPath, defaultSelectedTags }) => {
     const { t } = useTranslation();
-    const { tags, showMoreTags, handleShowMoreTagsChange } = useDisplayedTags(
-        useAllTags()
-    );
-    const [filteredTags, setFilteredTags] = useState([]);
+    const {
+        tagsData,
+        changeTagSelected,
+        isDisplayMoreTags,
+        changeDisplayMoreTags
+    } = useTags(defaultSelectedTags);
+    const [isEnableTags, setIsEnableTags] = useState(true);
```

Comment:
@hadasfarhi Please create a custom hook for that. View should be clearer as possible